addons/website/tests/test_website_visitor.py

```diff
@@ -287,40 +282,110 @@ def test_visitor_creation_on_tracked_page(self):
+            'name': 'Active Bob',
             'lang_id': self.env.ref('base.lang_en').id,
             'country_id': self.env.ref('base.be').id,
             'website_id': 1,
-            'partner_id': partner_demo.id,
-        })
-        self.assertTrue(old_visitor.active)
-        self.assertEqual(partner_demo.visitor_ids, old_visitor, "Visitor and its partner should be synchronized")
+            'last_connection_datetime': datetime.now() - timedelta(days=1)
+        }, {
+            'name': 'Client Richard',
+            'lang_id': self.env.ref('base.lang_en').id,
+            'country_id': self.env.ref('base.be').id,
+            'website_id': 1,
+            'partner_id': self.partner_demo.id,
+            'last_connection_datetime': datetime.now() - timedelta(days=15)
+        }])
 
-        # archive old visitor
-        old_visitor.last_connection_datetime = datetime.now() - timedelta(days=8)
-        self.env['website.visitor']._cron_archive_visitors()
-        self.assertEqual(old_visitor.active, False, "Visitor should be archived after inactivity")
+        self._test_clean_inactive_visitors(inactive_visitors, active_visitors)
 
-        # reconnect with new visitor.
-        self.url_open(self.tracked_page.url)
-        new_visitor = self._get_last_visitor()
-        self.assertFalse(new_visitor.partner_id)
-        self.assertTrue(new_visitor.id > old_visitor.id, "A new visitor should have been created.")
-        self.assertVisitorTracking(new_visitor, self.tracked_page)
+    def _test_clean_inactive_visitors(self, inactive_visitors, active_visitors):
+        """ This method will test that the visitors are correctly deleted when inactive.
+
+        - inactive_visitors: all visitors that should be unlinked by the CRON
+          '_cron_clean_inactive_visitors'
+        - active_visitors: all visitors that should NOT be cleaned because they are either active
+          or have some important data linked to them (partner, ...) and we want to keep them.
 
-        with self.mock_visitor_from_request(force_visitor=new_visitor):
-            self.authenticate('demo', 'demo')
-        (new_visitor | old_visitor).flush()
-        partner_demo.flush()
-        partner_demo.invalidate_cache(fnames=['visitor_ids'])
-        self.assertEqual(partner_demo.visitor_ids, old_visitor, "The partner visitor should be back to the 'old' visitor.")
+        We use this method as a private tool so that sub-module can also test the cleaning of visitors
+        based on their own sets of conditions. """
 
-        new_visitor = self.env['website.visitor'].search([('id', '=', new_visitor.id)])
-        self.assertEqual(len(new_visitor), 0, "The new visitor should be deleted when visitor authenticate once again.")
-        self.assertEqual(old_visitor.active, True, "The old visitor should be reactivated when visitor authenticates once again.")
+        WebsiteVisitor = self.env['website.visitor']
```

Comment:
Would add "active_test", just to be sure someone did not change our cron by an archive again :) . To be sure they are deleted.